### DISCUSSION TITLE
43. 웹 브라우저에서 URL을 입력했을 때 네트워크 계층에서는 어떤 일이 일어나나요?

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#44] [General]
.

**Comment by eun-byeol** [comment]
`IP 패킷을 생성`하고 `라우팅 및 전송`하는 과정이 일어납니다.

캡슐화 과정에서 전송 계층에서 받은 세그먼트에 `IP 헤더`를 추가하여 IP 패킷을 생성합니다. IP 헤더에는 송신지 IP와 목적지 IP를 포함됩니다. 이때 IP 주소는 같거나 서로 다른 네트워크 간 통신을 위해 사용됩니다.

생성된 IP 패킷은 `라우터`를 통해 전달됩니다. 라우터는 목적지 IP 주소를 확인하고 라우팅 테이블에 기반한 `최적 경로를 선택`하여 패킷을 전송합니다. 이때 패킷이 네트워크 경로의 최대 전송 단위(MTU)를 초과하면 단편화가 발생할 수 있습니다. 단편화된 패킷은 수신 측에서 재조립됩니다.

---
> 캡슐화 과정

![image](https://github.com/user-attachments/assets/60f5f44b-dcaf-483d-9086-9fce7da0468c)
